pkg/controllers/user/helm/common.go

```diff
@@ -163,8 +173,14 @@ func (l *Lifecycle) generateTemplates(obj *v3.App) (string, string, *common.Helm
 	}
 
 	// notes.txt
-	commands = append([]string{"template", appDir, "--name", obj.Name, "--namespace", obj.Spec.TargetNamespace, "--notes"}, setValues...)
-	cmd = exec.Command(helmName, commands...)
+	if obj.Status.HelmVersion == common.HELMNAMEV2 || obj.Status.HelmVersion == "" {
+		commands = append([]string{"template", appDir, "--name", obj.Name, "--namespace", obj.Spec.TargetNamespace, "--notes"}, setValues...)
+		cmd = exec.Command(common.HELMNAMEV2, commands...)
+	}
+	if obj.Status.HelmVersion == common.HELMNAMEV3 {
+		commands = append([]string{"upgrade", "--install", obj.Name, appDir, "--dry-run", "--namespace", obj.Spec.TargetNamespace, "--kubeconfig", tempDir.KubeConfigInJail}, setValues...)
```

Comment:
dry run ?

Reply:
helm 3 removed the template option, so there isn't a great way to create the notes.txt information

Reply:
the dry run is returning more info than --notes previously did, so I do need to refactor that code. Good catch!